src/lib/components/SnackbarContainer/index.js

```diff
@@ -34,24 +34,51 @@ class SnackbarContainer extends BaseElement {
 
   constructor() {
     super();
+    this.onBeforeInstallPrompt = this.onBeforeInstallPrompt.bind(this);
     this.onStateChanged = this.onStateChanged.bind(this);
   }
 
   connectedCallback() {
     super.connectedCallback();
     store.subscribe(this.onStateChanged);
     this.onStateChanged();
+
+    if (!this.acceptedCookies) {
+      window.addEventListener(
+        "beforeinstallprompt",
+        this.onBeforeInstallPrompt,
+      );
+    }
   }
 
   disconnectedCallback() {
     super.disconnectedCallback();
     store.unsubscribe(this.onStateChanged);
+    window.removeEventListener(
+      "beforeinstallprompt",
+      this.onBeforeInstallPrompt,
+    );
+  }
+
+  onBeforeInstallPrompt(e) {
+    e.preventDefault();
+    this.installPrompt = e;
```

Comment:
Is it worth checking to see if the user has already accepted cookies here? If they have, we could let the event proceed and if they haven't we could capture the event.

Reply:
This will never run if the user has accepted cookies. It's attached as a listener if line 46 is false. I just didn't want to attach the listener at all if they've accepted cookies.

Reply:
got it, sounds good